### PR TITLE
Deprecate formatted

### PR DESCRIPTION
**Pull request by randomboolean**
Deprecate FormattedWarpscriptFunction.

### CODE REVIEW

File: warp10/src/main/java/io/warp10/script/functions/shape/RESHAPE.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 //
-//   Copyright 2019  SenX S.A.S.
+//   Copyright 2019-2021  SenX S.A.S.
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/shape/RESHAPE.java:2]: Unnecessary.

File: warp10/src/main/java/io/warp10/script/functions/shape/SHAPE.java [hunk 2]
```diff
@@ -18,42 +18,50 @@
...
-  public static final String SHAPE = "shape";
-
-  private final Arguments args;
-  private final Arguments output;
-  protected Arguments getArguments() { return args; }
-  protected Arguments getOutput() { return output; }
+/**
+ * Return the shape of an input list if it could be a tensor (or multidimensional array), or raise an Exception.
+ * .addArgument(List.class, LIST, "The input list.")
+ * .addOptionalArgument(Boolean.class, FAST, "If true, it does not check if the sizes of the nested lists are coherent and it returns a shape based on the first nested lists at each level. Default to false.", false)
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/shape/SHAPE.java:30]: Rewrite comment to remove function calls.

**Review by ftence — CHANGES_REQUESTED** [review]
You could also update copyright notices in classes where you simply added `@Deprecated` but this is optional IMHO.

**Comment by randomboolean** [comment]
Note that the code is not fully retro-compatible with map input syntax.

File: warp10/src/main/java/io/warp10/script/functions/shape/PERMUTE.java [hunk 3]
```diff
@@ -28,9 +28,9 @@
 
 /**
  * Permute the dimensions of a nested LIST as if it were a tensor or a multidimensional array.
- * .addArgument(List.class, TENSOR, "The nested LIST for which dimensions will be permuted as if it were a tensor.")
- * .addListArgument(Long.class, PATTERN, "The permutation pattern (a LIST of LONG).")
- * .addOptionalArgument(Boolean.class, FAST, "If true, it does not check if the sizes of the nested lists are coherent before operating. Default to false.", false)
+ * - param TENSOR The nested LIST for which dimensions will be permuted as if it were a tensor.
+ * - param PATTERN The permutation pattern (a LIST of LONG)
+ * - param FAST If true, it does not check if the sizes of the nested lists are coherent before operating. Default to false.
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/shape/PERMUTE.java:33]: typo: "Defaults to..."

File: warp10/src/main/java/io/warp10/script/functions/shape/SHAPE.java [hunk 4]
```diff
@@ -26,8 +26,8 @@
 
 /**
  * Return the shape of an input list if it could be a tensor (or multidimensional array), or raise an Exception.
- * .addArgument(List.class, LIST, "The input list.")
- * .addOptionalArgument(Boolean.class, FAST, "If true, it does not check if the sizes of the nested lists are coherent and it returns a shape based on the first nested lists at each level. Default to false.", false)
+ * - param LIST The input list
+ * - param FAST If true, it does not check if the sizes of the nested lists are coherent and it returns a shape based on the first nested lists at each level. Default to false
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/shape/SHAPE.java:30]: typo: "Defaults to..."

**Review by ftence — CHANGES_REQUESTED** [review]

**Review by ftence — APPROVED** [review]

**Pull request merged**